Automattic/wp-calypso
# site-roles: migrate to use react query

#### Changes proposed in this Pull Request

* Create and use `useSiteRolesQuery` and `withSiteRoles`
* Move `getWpcomFollowerRole` to `state/sites/selectors`
* Remove all `site-roles` related Redux state

#### Testing instructions

- **Any site**:
  - Go to `/people/team/<site-slug>`
  - Click on any user that isn't you
  - Verify roles are shown as expected
- **Jetpack site**:
  - Go to `marketing/traffic/<site-slug>`
  - In the _Site stats_ section open the foldable card
  - Make sure roles are shown as expected
- **P2**:
  - Go to `/people/new/<site-slug>`
  - Verify roles are shown both in the _Invite_ section **and** the _Invite link_ section
  - Also make sure the last shown role is a _Viewer_ and no _Follower_ role
